typings/pins/audioout.d.ts

```diff
@@ -65,10 +67,12 @@ declare module 'pins/audioout' {
     public static readonly Silence: MixerSilence
   }
 
+  type AudioOutCallback = (value?: number) => void
```

Comment:
This is a small point. I'm not sure why the `value` argument is declared as optional here. The AudioOut implementation always passes the number argument. I'm still not very familiar with TypeScript though. Perhaps it is declared this way so that the a callback that does not use the `value` argument can be passed of the form `callback()`?

Reply:
Thanks for pointing that out. It is my mistake. I had overlooked that the second argument number of AudioOut.enque(AudioOut.callback) is [required](https://github.com/Moddable-OpenSource/moddable/blame/public/typings/pins/audioout.d.ts#L50). I pushed a fix.

>Perhaps it is declared this way so that the a callback that does not use the value argument can be passed of the form callback()?

This is actually achieved even if `value` argment is not optional. The below is a valid code.

```typescript
type FunctionThatTakesOneNumber = (value: number) => void
const myFunction: FunctionThatTakesOneNumber = (/* it doesn't take any argument */) => {  }
```

So even if the type definition stipulates that a function must have arguments, it is up to the user to decide whether or not to actually use them.

Reply:
OK, I understand. Functions may ignore input arguments but, when they use an argument, they must respect its type. That you for helping me find my way through TypeScript!

By the way, in applying the update I noticed that the second form of `mix` (recently added) is missing, so I will add that too. It is useful for optimizing off-line mixing, like some of the audio tests do.

```ts
public mix(buffer: BufferLike): void
```